bridges/bin/runtime-common/src/extensions/check_obsolete_extension.rs

```diff
@@ -267,83 +267,89 @@ macro_rules! generate_bridge_reject_obsolete_headers_and_messages {
 	($call:ty, $account_id:ty, $($filter_call:ty),*) => {
 		#[derive(Clone, codec::Decode, Default, codec::Encode, Eq, PartialEq, sp_runtime::RuntimeDebug, scale_info::TypeInfo)]
 		pub struct BridgeRejectObsoleteHeadersAndMessages;
-		impl sp_runtime::traits::SignedExtension for BridgeRejectObsoleteHeadersAndMessages {
+		impl sp_runtime::traits::TransactionExtensionBase for BridgeRejectObsoleteHeadersAndMessages {
 			const IDENTIFIER: &'static str = "BridgeRejectObsoleteHeadersAndMessages";
-			type AccountId = $account_id;
-			type Call = $call;
-			type AdditionalSigned = ();
-			type Pre = (
+			type Implicit = ();
+		}
+		impl<Context> sp_runtime::traits::TransactionExtension<$call, Context> for BridgeRejectObsoleteHeadersAndMessages {
+			type Pre = Option<(
 				$account_id,
 				( $(
 					<$filter_call as $crate::extensions::check_obsolete_extension::BridgeRuntimeFilterCall<
 						$account_id,
 						$call,
 					>>::ToPostDispatch,
 				)* ),
-			);
+			)>;
+			type Val = ();
 
-			fn additional_signed(&self) -> sp_std::result::Result<
-				(),
-				sp_runtime::transaction_validity::TransactionValidityError,
-			> {
-				Ok(())
-			}
-
-			#[allow(unused_variables)]
 			fn validate(
 				&self,
-				who: &Self::AccountId,
-				call: &Self::Call,
-				_info: &sp_runtime::traits::DispatchInfoOf<Self::Call>,
+				origin: <$call as sp_runtime::traits::Dispatchable>::RuntimeOrigin,
+				call: &$call,
+				_info: &sp_runtime::traits::DispatchInfoOf<$call>,
 				_len: usize,
-			) -> sp_runtime::transaction_validity::TransactionValidity {
+				_context: &mut Context,
+				_self_implicit: Self::Implicit,
+				_inherited_implication: &impl codec::Encode,
+			) -> Result<
+				(
+					sp_runtime::transaction_validity::ValidTransaction,
+					Self::Val,
+					<$call as sp_runtime::traits::Dispatchable>::RuntimeOrigin,
+				), sp_runtime::transaction_validity::TransactionValidityError
+			> {
+				use sp_runtime::traits::AsSystemOriginSigner;
 				let tx_validity = sp_runtime::transaction_validity::ValidTransaction::default();
-				let to_prepare = ();
+				let who = origin.as_system_origin_signer().ok_or(sp_runtime::transaction_validity::InvalidTransaction::BadSigner)?;
```

Comment:
> If the goal is to have a PR with minimal changes, then it is good to me.

Exactly this. Until we actually start using general transactions in our codebase, specifically bridges in this scenario, this is enforced to be a signed origin just to preserve previous functionality with minimal changes.

In the future, this can be changed to accept any origin (along with the relayer refund extension) and just ignore non-signed origins, since refunds don't really make sense without an account to refund to.